activerecord/lib/active_record/connection_adapters/sqlite3/database_statements.rb

```diff
@@ -114,6 +105,25 @@ def high_precision_current_timestamp
         end
 
         private
+          def internal_begin_transaction(mode, isolation)
+            if isolation
+              raise TransactionIsolationError, "SQLite3 only supports the `read_uncommitted` transaction isolation level" if isolation != :read_uncommitted
+              raise StandardError, "You need to enable the shared-cache mode in SQLite mode before attempting to change the transaction isolation level" unless shared_cache?
+            end
+
+            log("begin #{mode} transaction", "TRANSACTION") do
+              with_raw_connection(allow_retry: true, materialize_transactions: false) do |conn|
+                if isolation
+                  ActiveSupport::IsolatedExecutionState[:active_record_read_uncommitted] = conn.get_first_value("PRAGMA read_uncommitted")
```

Comment:
Ok, I missed this in review, sorry, but Is there a reason `IsolatedExecutionState` was used here? It looks more like something that should be an instance variable.

Reply:
Nevermind, it was there first. I'll blame deeper.